mmv1/third_party/terraform/services/bigquery/data_source_google_bigquery_tables.go

```diff
@@ -0,0 +1,143 @@
+package bigquery
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleBigQueryTables() *schema.Resource {
+
+	dsSchema := map[string]*schema.Schema{
+		"dataset_id": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The ID of the dataset containing the tables.",
+		},
+		"project": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The ID of the project in which the dataset is located. If it is not provided, the provider project is used.",
+		},
+		"tables": {
+			Type:     schema.TypeList,
+			Computed: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"labels": {
+						Type:     schema.TypeMap,
+						Computed: true,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+					},
+					"table_id": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Read:   DataSourceGoogleBigQueryTablesRead,
+		Schema: dsSchema,
+	}
+}
+
+func DataSourceGoogleBigQueryTablesRead(d *schema.ResourceData, meta interface{}) error {
+
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	datasetID := d.Get("dataset_id").(string)
+
+	project, err := tpgresource.GetProject(d, config)
+
+	if err != nil {
+		return fmt.Errorf("Error fetching project: %s", err)
+	}
+
+	params := make(map[string]string)
+	tables := make([]map[string]interface{}, 0)
+
+	for {
+		url := fmt.Sprintf("https://bigquery.googleapis.com/bigquery/v2/projects/%s/datasets/%s/tables", project, datasetID)
```

Comment:
Can you construct the `url` using base path like this instead of hardcoding the API endpoint? https://github.com/GoogleCloudPlatform/magic-modules/blob/40b53825878a3eaaf56d8dbcc49835c69723a31a/mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go#L1805

Reply:
Sounds good, lots of learning on my end with these suggestions so I really appreciate your patience with providing them.

Addressed!

Reply:
No problem, thanks for addressing the feedback. For this block let's also check `err` and return it if it's present. This error handling is based on golang's style guide: https://go.dev/blog/error-handling-and-go. Also the indentation here is different from the rest of the block.